spot_ros2_control/examples/src/joint_command_passthrough.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2024 Boston Dynamics AI Institute LLC. All rights reserved.
+
+// Note(llee): I think I would like to move this to `action_stack_spot` because so far it's pretty specific to Spot
+
+#include <chrono>
+#include <functional>
+#include <map>
+#include <memory>
+#include <ranges>
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joint_state.hpp"
+#include "spot_ros2_control/spot_joint_map.hpp"
+#include "std_msgs/msg/float64_multi_array.hpp"
+
+class JointCommandPassthrough : public rclcpp::Node {
+ public:
+  JointCommandPassthrough() : Node("joint_passthrough") {
+    joint_states_sub_ = create_subscription<sensor_msgs::msg::JointState>(
+        "Ethernet0/joint_states", 10,
+        std::bind(&JointCommandPassthrough::joint_states_callback, this,
+                  std::placeholders::_1));  // todo (llee): make this topic name a parameter
+
+    joint_commands_sub_ = create_subscription<sensor_msgs::msg::JointState>(
+        "Ethernet0/joint_commands", 10,
+        std::bind(&JointCommandPassthrough::joint_commands_callback, this,
+                  std::placeholders::_1));  // todo (llee): make this topic name a parameter
+
+    command_pub_ = create_publisher<std_msgs::msg::Float64MultiArray>(
+        "/Ethernet0/forward_position_controller/commands", 10);  // todo (llee): make this topic name a parameter
+
+    spot_command_.data.reserve(spot_ros2_control::kNjointsArm);
+  }
+
+ private:
+  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_states_sub_;
+  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_commands_sub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr command_pub_;
+  std_msgs::msg::Float64MultiArray spot_command_;
+
+  int count_ = 0;
+
+  void joint_states_callback(const sensor_msgs::msg::JointState& msg) {
+    // Keep track of the current joint positions
+    if (msg.name.size() != static_cast<std::vector<int>::size_type>(spot_ros2_control::kNjointsArm)) {
+      RCLCPP_ERROR(get_logger(), "Expected %i dofs, but got %li", spot_ros2_control::kNjointsArm, msg.name.size());
+      return;
+    }
+
+    spot_command_.data.clear();
+    spot_command_.data.resize(spot_ros2_control::kNjointsArm);
```

Comment:
Does the size change on a clear? Do we need this here or can we just do it once in the constructor?

Reply:
I _think_ clear resizes it to 0 and released the memory which is why I kept this here just to hold onto that memory, but I might be wrong

Reply:
I think `clear()` sets the size to 0 but doesnt free the memory

Reply:
Ok, so clear doesn't change the capacity (i.e. free the memory), but does set the size to 0. I think 

```suggestion
    std::fill(spot_command_.data.begin(), spot_command_.data.end(), 0);
```

will do what you want but faster.